### PR TITLE
mobile: moving file checks to their own file

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/mobile-perf.yml [hunk 1]
```diff
@@ -62,16 +62,8 @@ jobs:
...
-            source/common/listener_manager/listener_manager_impl.h
-            source/server/overload_manager_impl.cc
-            source/common/network/listen_socket_impl.h
-            source/common/network/tcp_listener_impl.h
-            source/server/guarddog_impl.h
-            source/server/watchdog_impl.h
-            source/server/options_impl.cc
-            source/extensions/access_loggers/common/file_access_log_impl.h
+            # TODO replace with tail -n + mobile/test/performance/files_em_does_not_use
+            rm `tail -n +2 mobile/test/performance/files_em_does_not_use`
```
Comment (alyssawilk) [inline .github/workflows/mobile-perf.yml:66]: I'm not sure if the mobile/ is needed here - I think we're not in the mobile directory which would make this correct but not sure.

Comment (phlax) [reply .github/workflows/mobile-perf.yml:66]: > I think we're not in the mobile directory

thats correct

Comment (phlax) [inline .github/workflows/mobile-perf.yml:66]:
```suggestion
            TOREMOVE=$(tail -n +2 mobile/test/performance/files_em_does_not_use)
            rm $TOREMOVE
```

current way would swallow any errors and backticks are strongly discouraged for exec in bash

even better would be to use something like the xargs way i posted offline (~relatedly) as that would even pass shellcheck

**Review by phlax — APPROVED** [review]
lgtm, thanks @alyssawilk

File: .github/workflows/mobile-perf.yml [hunk 2]
```diff
@@ -62,16 +62,8 @@ jobs:
...
-            source/common/http/route_config_update_requster.h
-            source/common/listener_manager/listener_manager_impl.h
-            source/server/overload_manager_impl.cc
-            source/common/network/listen_socket_impl.h
-            source/common/network/tcp_listener_impl.h
-            source/server/guarddog_impl.h
-            source/server/watchdog_impl.h
-            source/server/options_impl.cc
-            source/extensions/access_loggers/common/file_access_log_impl.h
+            TOREMOVE=$(tail -n +2 mobile/test/performance/files_em_does_not_use)
```
[the remark this answers is not in the document]
Comment (RyanTheOptimist) [reply .github/workflows/mobile-perf.yml:65]: That seems like a wise idea.

**Review by phlax — APPROVED** [review]

**Pull request merged**